lib/AST/NameLookup.cpp

```diff
@@ -2210,12 +2212,14 @@ ClassDecl *SuperclassDeclRequest::evaluate(Evaluator &evaluator,
     // Find the inherited declarations referenced at this position.
     auto inheritedTypes =
       evaluator(InheritedDeclsReferencedRequest{subject, i});
+    if (!inheritedTypes)
+      continue; 
```

Comment:
Interesting question here: as a matter of policy, should we bail out when we encounter a cycle, or continue on? Do we need a way to distinguish between "a cycle happened but we can still try to compute something useful" and "something else (like cancellation) happened and we should return immediately" in some consistent way?

Reply:
I went ahead and made the change simpler. I wasn't 100% sure this version was NFC, but this version should be.